projects/igniteui-angular/src/lib/accordion/accordion.component.ts

```diff
@@ -256,6 +269,16 @@ export class IgxAccordionComponent implements AfterContentInit, AfterViewInit, O
         this.panels.forEach(panel => panel.collapse());
     }
 
+    private collapseAllExceptLast(): void {
+        const lastExpanded = this.panels?.filter(p => !p.collapsed && !p.header.disabled).pop();
+        this.panels?.forEach((p: IgxExpansionPanelComponent) => {
+            if (p !== lastExpanded && !p.header.disabled) {
+                p.collapsed = true;
+                p.cdr.detectChanges();
```

Comment:
One call to `cdr.detectChanges()` after collapsing all panels would be better than calling `detectChanges()` on each collapsed panel. Don't reference each expansion panel `cdr` but instead call detectChanges on a cdr from the accordion itself after the `forEach` loop.